File: ios/Kurir/modules/auth/AuthDelegate.swift
//
//  AuthDelegate.swift
//  Kurir
//
//  Created by Loren on 19/8/19.
//  Copyright © 2019 Crazybean Studio. All rights reserved.
//

import Foundation
import MobileSDK
import Mobilex

class AuthDelegate: Delegate<AuthView, AuthViewModel> {
    override func onCreate(_ params: [String : Any?]? = nil) {
        view?.showLogin(auth: nil)
    }
}

File: android/src/main/java/au/com/crazybean/mobilex/kurir/modules/auth/AuthDelegate.kt
package au.com.crazybean.mobilex.kurir.modules.auth

import android.os.Bundle
import androidx.lifecycle.Observer
import au.com.crazybean.mobilex.modules.auth.AuthView
import au.com.crazybean.sdk.mvvm.Delegate

class AuthDelegate(view: AuthView?,
                   viewModel: AuthViewModel) : Delegate<AuthView, AuthViewModel>(view, viewModel) {

    fun onLoadClick() {
        viewModel.login("Loren", "1234")
                .observe(this, Observer {
                    view?.showLogin(it)
                })
    }

    override fun onCreate(params: Bundle) {
        super.onCreate(params)
        view?.showLogin(null)
    }
}